hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -264,6 +269,162 @@ public void renameTable(TableIdentifier from, TableIdentifier originalTo) {
     }
   }
 
+  @Override
+  public boolean viewExists(TableIdentifier identifier) {
+    return HiveCatalogUtil.isTableWithTypeExists(clients, identifier, TableType.VIRTUAL_VIEW);
```

Comment:
What is the rationale for storing an Iceberg view as a VIRTUAL_VIEW type? Shouldn't we just use a different `table_type` property like `iceberg-view`?

Reply:
I think we started this [discussion](https://github.com/apache/iceberg/issues/8698#issuecomment-1770264064) but never came to any conclusion . I had only one concerns for adding it as property for backward-compatibility. 
There must be a check at all the places with table property whether it is as Iceberg-table or iceberg-view. We do validate with [validateTableIsIceberg](https://github.com/apache/iceberg/blob/99843f03efecc3f5ef7d3e1d32aae5eff22cb315/hive-metastore/src/main/java/org/apache/iceberg/hive/HiveOperationsBase.java#L96) for table. Please suggest.

Reply:
@pvary , Please share your thoughts on this.

Reply:
I think using VIRTUAL_VIEW is more Hiveish solution, but I see the merits of using the already existing Iceberg property key with different value.
Since the original check already does an exact matching for 'iceberg', adding a new value would only cause issues for users who do their own filtering without the Iceberg code. So I do not have a strong opinion, we can use Ryan's suggestion to move forward.

Reply:
I will go ahead with adding `tableType` as `External_Table` and introducing new properties as `table_type` with value `iceberg-view`

Reply:
@rdblue , I have added a new [prop](https://github.com/apache/iceberg/pull/8907/files#diff-1ae8e9490fe1a4b6da8842c1c313fa57e68a674e186bfeace1c763bee1381faaR49)  for view. 

Please help me to resolve this as per community sync discussions. 
Currently I am setting up both table type as VIRTUAL_VIEW and table prop as ICEBERG-VIEW.